adapter/inventory.bzl

```diff
@@ -0,0 +1,27 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+def _inventory_gen(name, packages, out):
+  args = ""
+  for k, v in packages.items():
+    args += "-p %s:%s " % (k, v)
+  
+  native.genrule(
+      name = name+"_gen",
+      outs = [out],
```

Comment:
in this case, the tool itself just dumps to stdout.  the genrule `cmd` redirects it to a file.  this was done for expediency/simplicity.  it should be very easy to add the logic around passing in a file stream to the command, if you think that is clearer.